### ISSUE TITLE
Social Registration no longer respecting assigned user role for form

### ISSUE THREAD

**Issue by mrbm** [ultimatemember/ultimatemember#203]
This used to be working but no longer is:

I have a form Called: Social Login Expert

That form is has: Assign Role -> Expert

When when people register with that login via linked in, instead of that user being assigned to expert, they are assigned to a role of: Community Member (which is in the ultimate member settings for "Default New User Role"

So...its broke.

Guessing something changed in Ultimate Member. Not sure how long ago. Don't trust this data, but I believe the last successful registration with the properly assigned role was:

19 Aug 2016 

Then had another user sign up on 

18 Oct 2016

When I noticed the issue.

Let me know what you think please. Thanks.

**Comment by mrbm** [comment]
Follow up, using a social registration pre-release from July 26th that you sent me. Thanks.

**Comment by champsupertramp** [comment]
hi @mrbm, can you please create a ticket on our support page so one of my team mates can confirm this issue and send you the latest pre-release version? thanks

**Comment by champsupertramp** [comment]
hi @mrbm, can you please create a ticket on our support page so one of my team mates can confirm this issue and send you the latest pre-release version? thanks

**Comment by champsupertramp** [comment]
hi @mrbm, can you please create a ticket on our support page so one of my team mates can confirm this issue and send you the latest pre-release version? thanks

**Comment by champsupertramp** [comment]
hi @mrbm, can you please create a ticket on our support page so one of my team mates can confirm this issue and send you the latest pre-release version? thanks

**Comment by champsupertramp** [comment]
hi @mrbm, can you please create a ticket on our support page so one of my team mates can confirm this issue and send you the latest pre-release version? thanks

**Comment by mrbm** [comment]
email sent and I documented the whole issue with commented screenshots

**Comment by mrbm** [comment]
hey no response from the support with the new social registration...deadline for conference coming this sat...and would still like to have time to do real testing....

**Comment by mrbm** [comment]
As quick work around I'm going to add the following to the template of the first page the form redirects to, if there is any issues with this code, please let me know!

```
global $ultimatemember;
$ultimatemember->user->set( $current_user_id  );
if ( current_user_can('manage_options') || ( !um_user('super_admin') || um_user('role') != 'admin' ) ) {
                $new_role = 'expert';
                $ultimatemember->user->set_role( $new_role );
}
```

**Comment by chriseckman** [comment]
I just tested this on my site and I can confirm that the role is defaulting to member regardless of what is filled out in the registration.

**Comment by mrbm** [comment]
@chriseckman thanks for confirming that. The trusted @champsupertramp I leave it in your hands.

**Comment by chriseckman** [comment]
I found this in the docs that might explain what is going on, but it's kinda hard to follow. 

http://docs.ultimatemember.com/article/223-social-login-assigning-roles

It states:

> "When a user registers, the form on overlay will use a community role based from global default role set in Ultimate Member > Settings > Users > Default New User Role that will be assigned to the user on successful registration."

So it looks like it will set the role to the default... which is what it is doing. But later says:

> "The other way is using Social Login Shortcodes. You can put shortcode anywhere on the page or you can add a content block field in Login/Registration form where you can place  the social login shortcode. To create a Social Login Shortcode, go to Ultimate Member > Settings > Social Login> Add New"

So does this override the 2-step process of the social login authentication and then registration form? So if we put a shortcode for the social login at the bottom of the normal registration page this will work?

**Comment by mrbm** [comment]
@chriseckman 

I have a default role set which it is using.

However I also have a different role set specifically for a social registration form. When users use that social registration form...its not respecting the special social registration forms setting. Thats the bug.

Used to work a little while back...so something changed in the ultimate member core to make it no longer work.

**Comment by champsupertramp** [comment]
Hi @mrbm  @chriseckman 

I'm working on this issue.  I'll keep you guys updated.

Regards,

**Comment by chriseckman** [comment]
@champsupertramp Any updates on this one?

**Comment by champsupertramp** [comment]
Hi @chriseckman / @mrbm ,

Can you please try the updated pre-release version of UM core in the link?
https://github.com/ultimatemember/ultimatemember/releases

We've fixed the issue in assigning of role on register process.  I'll send the pre-release version of UM Social Login in your paid support tickets.

Regards,
